src/raven.js

```diff
@@ -469,6 +469,30 @@ Raven.prototype = {
         return this;
     },
 
+    /*
+     * Adds a shouldSendCallback callback that chains after any previously set
+     * callback(s).
+     *
+     * @param {function} callback The callback to run which allows
+     *                            introspecting the blob before sending
+     * @return {Raven}
+     */
+    addShouldSendCallback: function(callback) {
+        var prevCallback = this._globalOptions.shouldSendCallback;
+        if (prevCallback) {
+            this._globalOptions.shouldSendCallback = function() {
+                if (prevCallback && !prevCallback.apply(null, arguments)) {
```

Comment:
I don't think it's necessary to check `prevCallback` again. So I think this can be written as:

```javascript
return prevCallback.apply(null, arguments) && callback.apply(null, arguments);
```